### PR TITLE
[FormRecognizer] Tests: supporting forms with blank pages

### PR DESCRIPTION
**Pull request by kinelski**
Filling the final gaps in our tests. Adding test scenarios for forms with blank pages. A few of them are being ignored because of service-side bugs.

There was no corresponding issue to close.

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 1]
```diff
@@ -464,16 +545,92 @@ public async Task StartRecognizeCustomFormsWithLabelsCanParseMultipageForms(bool
...
+                var formPage = recognizedForm.Pages[pageIndex];
+                var sampleLine = formPage.Lines[3];
+                var expectedText = pageIndex == 0 ? "Bilbo Baggins" : "Frodo Baggins";
+
+                Assert.AreEqual(expectedText, sampleLine.Text);
             }
         }
 
+        [Test]
+        public async Task StartRecognizeCustomFormsWithLabelsCanParseBlankPage()
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:567]: Currently there are 4 operations we are testing: content recognition, receipts recognition, custom form recognition without labels and custom form recognition with labels. For each one of them we have a "multipage form with a blank page" test.

The custom form recognition with labels, however, behaves differently from others because it treats the whole file as a single multipage form, while other operations parse each page individually. In order to test an actual "blank form" scenario, I've added a special case which makes use of the `blank.pdf` file, a single blank page. Just wanted to make sure that having a complete blank form in the custom form with labels scenario wouldn't break our client with unexpected responses from the service.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 2]
```diff
@@ -556,14 +713,12 @@ public async Task StartRecognizeCustomFormsWithoutLabels(bool useStream)
         }
 
         [Test]
-        [TestCase(true, true)]
-        [TestCase(true, false)]
-        [TestCase(false, true, Ignore = "Service returning 'Unsupported media type' error.")]
-        [TestCase(false, false, Ignore = "Service returning 'Unsupported media type' error.")]
-        public async Task StartRecognizeCustomFormsWithoutLabelsCanParseMultipageForms(bool useStream, bool includeTextContent)
+        [TestCase(true)]
+        [TestCase(false, Ignore = "Service returning 'Unsupported media type' error.")]
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:717]: Service-side issue. Service and SDK teams are already aware of it.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 3]
```diff
@@ -606,6 +761,58 @@ public async Task StartRecognizeCustomFormsWithoutLabelsCanParseMultipageForms(b
             }
         }
 
+        [Test]
+        [Ignore("Service bug: information about the blank page is not being returned.")]
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:765]: Another service-side issue. Service and SDK teams are already aware of it.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Infrastructure/FormRecognizerTestEnvironment.cs [hunk 4]
```diff
@@ -51,6 +51,12 @@ public FormRecognizerTestEnvironment() : base("formrecognizer")
         /// <summary>The name of the PDF file which contains the multipage form to be used for tests.</summary>
         private const string MultipageFormFilename = "multipage_invoice_noblank.pdf";
 
+        /// <summary>The name of the PDF file which contains the multipage form with a blank page to be used for tests.</summary>
+        private const string MultipageBlankFormFilename = "multipage_invoice1.pdf";
+
+        /// <summary>The name of the PDF file which contains the single-page form with a blank page to be used for tests.</summary>
+        private const string BlankFormFilename = "blank.pdf";
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Infrastructure/FormRecognizerTestEnvironment.cs:58]: Mariana and I have discussed a plan to move these to another file and keep the `FormRecognizerTestEnvironment` organized, but that will be done in another PR.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizedForm.cs [hunk 5]
```diff
@@ -32,7 +32,13 @@ internal RecognizedForm(DocumentResult_internal documentResult, IReadOnlyList<Pa
...
 
             PageRange = new FormPageRange(documentResult.PageRange[0], documentResult.PageRange[1]);
-            Fields = ConvertSupervisedFields(documentResult.Fields, readResults);
+
+            // documentResult.Fields is required and not null, according to the swagger file, but it's not
+            // present when a blank receipt is submitted for recognition.
+
+            Fields = documentResult.Fields == null
+                ? new Dictionary<string, FormField>()
+                : ConvertSupervisedFields(documentResult.Fields, readResults);
```
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizedForm.cs:41]: ohh interesting. Is this something the service will fix?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Infrastructure/FormRecognizerTestEnvironment.cs (hunk 4 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Infrastructure/FormRecognizerTestEnvironment.cs:58]: Thanks for proposing this Caio!!!

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 2 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:717]: out curiosity, is this something the service team will fix? do we know the ETA?

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:717]: It will be good to track this `Ignore` test in an issue, so later we don't forget to check with the service team the status of it and enable the tests again

**Review by maririos — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizedForm.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizedForm.cs:41]: I'm not sure if they're aware of this issue. This is one of the things I've been wanting to ask the SDK team about but didn't have the chance.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs (hunk 2 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:717]: > is this something the service team will fix?

Not as far as we know. There's a thread for that in the Form Recognizer service channel. According to the service team, this is a GitHub issue.
```
GitHub is not responding with a correct Content-Type
Content-Type: application/octet-stream
```

It's still a bit weird to understand what's going on since this only happens with custom recognition **without** labels. Recognition with labels works just fine.

I'll open issues for the ignored tests once I merge this PR. I'll add them to our backlog milestone.

Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:717]: #12319

Comment (annelo-msft) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:717]: > It's still a bit weird to understand what's going on since this only happens with custom recognition without labels. Recognition with labels works just fine.

Do we know why this is?  It would be worth understanding the difference between the endpoints so we can communicate this to the service team.

Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:717]: > Do we know why this is?
We don't, but they're aware. Krista has mentioned it in the FR channel. I'll send you a link to the discussion.
